source/MaterialXGenOgsXml/OgsXmlGenerator.cpp

```diff
@@ -180,13 +199,14 @@ namespace
     }
 }
 
+const string OgsXmlGenerator::OUTPUT_NAME = "out";
 const string OgsXmlGenerator::SAMPLER_SUFFIX = "Sampler";
 
 OgsXmlGenerator::OgsXmlGenerator()
 {
 }
 
-void OgsXmlGenerator::generate(const Shader* glsl, const Shader* hlsl, std::ostream& stream)
+void OgsXmlGenerator::generate(const std::string& shaderName, const Shader* glsl, const Shader* hlsl, std::ostream& stream)
```

Comment:
We generate with a supplied name to ensure that the XML fragment name is set to a desired value vs (shader->getName()).